### PR TITLE
Docs/Deployent and Data Tier Sizing

### PR DESCRIPTION
**Pull request by evantahler**
This enhances the `4. Deployment` section in a few ways

1. Adds an index page to the Deployment Section
1. Adds a new section to the Network Topology section about sizing the data tier
1. Moves the Environment Variables, Secrets, and xxx pages from the Support to Deployment section
1. Moves the list of example projects to a sub-page

Screen Shot 2022-01-24 at 5 55 03 PM

This PR also removes the `Debugging` page, because it addresses problems must users are not seeing.

### Development

- [ ] Application changes have been tested appropriately

### Impact

- [ ] Code follows company security practices and guidelines
- [ ] Security impact of change has been considered
- [ ] Performance impact of change has been considered
- [ ] Possible migration needs considered (model migrations, config migrations, etc.)

Please explain any security, performance, migration, or other impacts if relevant:

### Code review

- [ ] Pull request has a descriptive title and context useful to a reviewer. Screenshots or screencasts are attached where applicable.
- [ ] Relevant tags have been added to the PR (bug, enhancement, internal, etc.)

### CODE REVIEW

File: pages/docs/deployment/network.mdx [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+  source="Grouparoo Topology Simple"
+  alt="Grouparoo Topology Simple"
+  src="docs/deployment/grouparoo-deployment-topology-simple.png"
+  width={820}
+  height={552}
+/>
+
+The Application tier can be as simple as a single instance of Grouparoo, or you can scale the application horizontally by adding more instances. An "instance" in this case is a single node.js process or docker image which runs the Grouparoo application. In the simplest case, each instance is both processing background jobs (WORKER) and serving the website and API requests (WEB_SERVER). This is configured via [environment variables](/docs/deployment/environment).
+
+## Sizing the Grouparoo Data Tier
```
Comment (evantahler) [inline pages/docs/deployment/network.mdx:28]: @andyjih this new section talks about tuning your data tier

File: pages/docs/deployment/environment.mdx [hunk 2]
```diff
@@ -57,6 +57,15 @@ You can also modify the SSL connection options to your postgres database with th
 
 To use SQLite, set `DATABASE_URL="sqlite://grouparoo_development.sqlite"` which will be stored in your application directory by default, or you can provide an absolute path, ie: `DATABASE_URL="sqlite:///path/to/db.sqlite"`.
 
+## Batch Size Environment Variables
```
Comment (evantahler) [inline pages/docs/deployment/environment.mdx:60]: This new section talks about DB batch sizes

File: pages/docs/deployment/network.mdx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+
+### Postgres
+
+The Grouparoo application is **write-heavy**, making heavy use of a Postgres database to normalize and store copies of your data. We do this for a number of reasons, primarily:
+
+- To support the joining of Profile Properties for Records from multiple sources
```
Comment (andyjih) [inline pages/docs/deployment/network.mdx]:
```suggestion
- To support the joining of Record Properties for Records from multiple sources
```

File: pages/docs/deployment/network.mdx [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+
+To that end, the Postgres database you assign Grouparoo will need to be:
+
+1. Large enough to hold all the data you intend to import
+2. Fast enough to quickly read and write data at the speed it is changing.
+
+While a number of factors are at play when determining how "fast" to make your database, the following tips may help:
+
+- The number one reason your Grouparoo application may be working slowly is a slow disk. Cloud vendors are notoriously stingy with the "hard drives" they provide by default on cloud databases. For example if your database is an AWS RDS instance, we recommend attaching at least 100GB of **_allocated storage_**. AWS scales IOPS along with the allocated storage.
+- Ensure that you have enough Memory to keep a days's worth of data in working memory. For 1 million Records x 20 Properties, have at least 8GB of Ram allocated.
```
Comment (andyjih) [inline pages/docs/deployment/network.mdx]:
```suggestion
- Ensure that you have enough Memory to keep a day's worth of data in working memory. For 1 million Records x 20 Properties, have at least 8GB of Ram allocated.
```

**Review by andyjih — APPROVED** [review]
All of this content looks great! I also in favor of including a section that sets expectations on speed for an initial run vs. subsequent runs like you mentioned in tracker.

**Comment by evantahler** [comment]
Ok! New performance section added with the help of @pedroslopez

File: pages/docs/deployment/monitoring-and-performance.mdx [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+```
+
+If your Grouparoo sync is going slower than the above example, here are suggestions to speed things up. These are listed in order of likelihood to help.
+
+1. Ensure that your Sources are adequately provisioned for repeated large reads. Check your logs, query times, and vendor credits.
+2. [Ensure the Grouparoo DB is fast enough](/docs/deployment/network#postgres)
+3. [Slowly increase the Grouparoo Batch Sizes](/docs/deployment/environment#batch-size-environment-variables)
+4. Check if Grouparoo is hitting rate limits when exporting Records to your Destinations. Is there a larger plan available with more throughput?
+5. Slowly increase the number if [`WORKERS`](/docs/deployment/environment#required-environment-variables) "threads" the Grouparoo application uses. The number of workers can be thought of as internal processing "threads", each working in parallel to import and export your data. Increasing the number of `WORKERS` may speed up Grouparoo, provided your sources and destinations can handle the increased volume and load. If not, increasing the number of `WORKERS` will actually slow things down. An upper limit of 20 WORKERS per server should be maintained.
```
Comment (andyjih) [inline pages/docs/deployment/monitoring-and-performance.mdx]:
```suggestion
5. Gradually increase the number of [`WORKERS`](/docs/deployment/environment#required-environment-variables) "threads" the Grouparoo application uses. The number of workers can be thought of as internal processing "threads", each working in parallel to import and export your data. Increasing the number of `WORKERS` may speed up Grouparoo, provided your sources and destinations can handle the increased volume and load. If not, increasing the number of `WORKERS` might actually slow things down. An upper limit of 20 WORKERS per server should be maintained.
```

**Review by andyjih — APPROVED** [review]
Some minor tweaks, but looks good. Thanks!

**Pull request merged**